### PR TITLE
Fix EOL issue

### LINKED ISSUE #11: gulp-terser-js assumes that OS EOL is the file's EOL
Encountered this with gulp-terser-js@5.0.0 on Windows, but not Linux:

```
[10:13:19] TypeError: Cannot read property 'replace' of undefined
    at printError (C:\myproject\node_modules\gulp-terser-js\bin\index.js:71:69)
    at DestroyableTransform._transform (C:\myproject\node_modules\gulp-terser-js\bin\index.js:42:19)
    at DestroyableTransform.Transform._read (C:\myproject\node_modules\gulp-terser-js\node_modules\readable-stream\lib\_stream_transform.js:177:10)
    at DestroyableTransform.Transform._write (C:\myproject\node_modules\gulp-terser-js\node_modules\readable-stream\lib\_stream_transform.js:164:83)
    at doWrite (C:\myproject\node_modules\gulp-terser-js\node_modules\readable-stream\lib\_stream_writable.js:405:139)
    at writeOrBuffer (C:\myproject\node_modules\gulp-terser-js\node_modules\readable-stream\lib\_stream_writable.js:394:5)
    at DestroyableTransform.Writable.write (C:\myproject\node_modules\gulp-terser-js\node_modules\readable-stream\lib\_stream_writable.js:303:11)
    at DestroyableTransform.ondata (C:\myproject\node_modules\readable-stream\lib\_stream_readable.js:619:20)
    at DestroyableTransform.emit (events.js:210:5)
    at DestroyableTransform.EventEmitter.emit (domain.js:499:23)
[10:13:19] 'build' errored after 8.42 s
```

That LoC is:
```javascript
const pos = (more + 2) + fileContent.split(os.EOL)[error.line - 1].replace(/[^\t]/g, '').length * 3 + parseInt(col)
```

Replacing both instances of `os.EOL` in `node_modules/gulp-terser-js/bin/index.js` with `'\n'` fixes the problem in Windows so we get a nice error report.
Replacing `os.EOL` with `'\r'` in Linux makes the same problem reproducible in Linux.

We have a lot of files with `\n` as the line ending. What I suspect is that `terser` is reporting line numbers correctly, but `gulp-terser-js` is not correlating that to the file because it is assuming that `os.EOL` is the same line ending that the file is using.

Just getting this information out there before I have time to confirm my theory, write a test, or submit an MR.

### PR DESCRIPTION
**Pull request by A-312**
Fix #11 

# Q/A : 

Try to build js files with linux eol on Windows.

```json5
{
  "devDependencies": {
    "gulp-teser-js": "git://github.com/A-312/gulp-teser-js.git#fix11"
  }
}
```

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ function gulpTerser(options) {
       res.error.filePath = file.path
       res.error.fileContent = (typeof build === 'string') ? build : build[res.error.filename]
 
+      // use system new line instead terser new line
+      res.error.fileContent = res.error.fileContent.replace(/\n\r\u2028\u2029/g, os.EOL)
```
Comment (shark-horse) [inline bin/index.js]: I think that should be e.g.: `"gulp-terser-js": "https://github.com/A-312/gulp-terser-js.git#fix-11"`

That regex only matches those 4 characters in a row in that order and so makes no difference in the cases I tested.

This works in my tests:
`res.error.fileContent = res.error.fileContent.replace(/\r\n|\r|\n|\u2028|\u2029/g, os.EOL)`
The `|` means "or" so it matches any of the 4 characters individually, but it accounts for `\r\n` being counted as a single return. What I don't know 100% is if some regex find-and-replace libs would get into an infinite loop if they replace e.g. a `\n` with a `\r\n` and don't advance past the `\n` they insert, but this did not happen when I tested.

I tested 1 file with `\n` line endings on a Linux machine and a Windows machine, and the same file with the line endings changed to `\r\n` on both machines. In all 4 cases, the results were unchanged by this PR (`TypeError` on Windows with `\n`), but working with the LoC I submitted above (showed the JS compilation error text as expected with the correct line number); gulp-terser-js did not throw an unexpected error itself.)

**Comment by A-312** [comment]
That should be good

**Comment by shark-horse** [comment]
Most recent change works for me. :+1:

**Comment by A-312** [comment]
I will merge this evening

**Comment by shark-horse** [comment]
Thanks!

**Comment by A-312** [comment]
@shark-horse 5.0.1 is on npm

**Pull request merged**